src/Passes/Source/Rules/Factory.cpp

```diff
@@ -502,6 +507,82 @@ namespace quantum
              replace_branch_positive});
     }
 
+    void RuleFactory::optimiseSelectQuantumOne()
+    {
+        auto replace_select_positive = [](Builder& builder, Value* val, Captures& cap, Replacements& replacements) {
+            auto result = cap["result"];
+            auto cond   = llvm::dyn_cast<llvm::Instruction>(cap["cond"]);
+            // Replacing result
+            auto orig_instr = llvm::dyn_cast<llvm::Instruction>(val);
+            if (orig_instr == nullptr)
+            {
+                return false;
+            }
+
+            auto                      module   = orig_instr->getModule();
+            auto                      function = module->getFunction("__quantum__qir__read_result");
```

Comment:
After this line there is a risk (in the future) of name conflict between variable `function` and [`std::function`](https://cplusplus.com/reference/functional/function/).

Reply:
No that should not be an issue given that the cpp file does not use namespace std.